acq4/util/igorpro.py

```diff
@@ -172,6 +204,95 @@ def getVariable(self, folder, varName):
                 return r
 
 
+class ZMQIgorBridge(object):
+    """Bridge to Igor via ZMQ DEALER/ROUTER."""
+    _context = zmq.Context()
+
+    _types = {"NT_FP32": np.float32,
+              "NT_FP64": np.float64}
+
+    def __init__(self, host="tcp://localhost", port=5670):
+        super(ZMQIgorBridge, self).__init__()
+        self._unresolvedFutures = {}
+        self._currentMessageID = 0
+        self.address = "{}:{}".format(host, port)
+        self._socket = self._context.socket(zmq.DEALER)
+        self._socket.setsockopt(zmq.IDENTITY, "igorbridge")
+        self._socket.setsockopt(zmq.SNDTIMEO, 1000)
+        self._socket.connect(self.address)
+        self._poller = zmq.Poller()
+        self._poller.register(self._socket, zmq.POLLIN)
+        self._pollTimer = QtCore.QTimer()
+        self._pollTimer.timeout.connect(self._checkRecv)
+        self._pollTimer.start(100)
+
+    def __call__(self, cmd, *args, **kwds):
+        # TODO: Handle optional values whenever they become supported in Igor
+        messageID = self._getMessageID()
+        future = concurrent.futures.Future()
+        call = self.formatCall(cmd, params=args, messageID=messageID)
+        try:
+            self._socket.send_multipart(call)
+            self._unresolvedFutures[messageID] = future
+        except zmq.error.Again:
+            self._unresolvedFutures.pop(messageID)
+            future.set_exception(IgorCallError("Send timed out",
+                IgorCallError.TIMEDOUT))
+        return future
+
+    def _checkRecv(self):
+        socks = self._poller.poll(100)
```

Comment:
I'll test with receive timeout set to 0 and if that looks like it is still functional we'll go with that, since I don't really like having the poller either.